src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Apprenticeship.cs

```diff
@@ -144,5 +151,77 @@ public DraftApprenticeship CreateCopyForChangeOfParty(ChangeOfPartyRequest chang
 
             return result;
         }
+
+        private bool IsWaitingToStart(ICurrentDateTime currentDateTime)
+        {
+            return StartDate.Value > new DateTime(currentDateTime.UtcNow.Year, currentDateTime.UtcNow.Month, 1);
+        }
+
+        internal void ValidateApprenticeshipForStop(DateTime stopDate, long accountId, ICurrentDateTime currentDate)
+        {
+            if (PaymentStatus == PaymentStatus.Completed || PaymentStatus == PaymentStatus.Withdrawn)
+            {
+                throw new DomainException(nameof(PaymentStatus), "Apprenticeship must be Active or Paused. Unable to stop apprenticeship");
+            }
+
+            if (Cohort.EmployerAccountId != accountId)
+            {
+                throw new DomainException(nameof(accountId), $"Employer {accountId} not authorised to access commitment {Cohort.Id}, expected employer {Cohort.EmployerAccountId}");
+            }
+
+            if (IsWaitingToStart(currentDate))
+            {
+                if (stopDate.Date != StartDate.Value.Date)
+                    throw new DomainException(nameof(stopDate), "Invalid stop date. Date should be value of start date if training has not started.");
+            }
+            else
+            {
+                /// When asking for a stop date, only a month and year are provded by the UI, The day is not supplied.
+                /// As a result, when constructing comparisons, it is clear the dates must also be of the same format.
+                if (stopDate.Date > new DateTime(currentDate.UtcNow.Year, currentDate.UtcNow.Month, 1))
+                {
+                    throw new DomainException(nameof(stopDate), "Invalid Stop Date. Stop date cannot be in the future.");
+                }
+
+                if (stopDate.Date < new DateTime(StartDate.Value.Year, StartDate.Value.Month, 1))
+                {
+                    throw new DomainException(nameof(stopDate), "Invalid Stop Date. Stop date cannot be before the apprenticeship has started.");
+                }
+            }
+        }
+
+        private const DataLockErrorCode CourseChangeErrors = DataLockErrorCode.Dlock03 | DataLockErrorCode.Dlock04 | DataLockErrorCode.Dlock05 | DataLockErrorCode.Dlock06;
+        private bool IsCourseChangeError(DataLockErrorCode errorCode) => (errorCode & CourseChangeErrors) > 0;
+
+        internal void StopApprenticeship(DateTime stopDate, bool madeRedundant, UserInfo userInfo, Party party)
+        {
+            StartTrackingSession(UserAction.StopApprenticeship, party, Cohort.EmployerAccountId, Cohort.ProviderId, userInfo);
+
+            ChangeTrackingSession.TrackUpdate(this);
+
+            PaymentStatus = PaymentStatus.Withdrawn;
+            StopDate = stopDate;
+            MadeRedundant = madeRedundant;
+
+            IEnumerable<DataLockStatus> dataLocks;
```

Comment:
For readability, I'd pop this data lock stuff into a method, just so it can be given a name for clarity.

Reply:
Refactored into a method 👍